### ISSUE TITLE
BE crashes when filtering by a timestamp column from external Iceberg table

### ISSUE THREAD

**Issue by pedrong** [StarRocks/starrocks#29264]
While querying an Iceberg table from a Glue external catalog and using a timestamp column as a condition is crashing BE process with the error below:

```
*** Aborted at 1692037039 (unix time) try "date -d @1692037039" if you are using GNU date ***
PC: @          0x3074337 starrocks::FixedLengthColumnBase<>::get()
*** SIGSEGV (@0x0) received by PID 408 (TID 0x7f2392763700) from PID 0; stack trace: ***
    @          0x63375c2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f242f5db8e0 (unknown)
    @          0x3074337 starrocks::FixedLengthColumnBase<>::get()
    @          0x5954288 starrocks::parquet::FileReader::_filter_group()
    @          0x5955167 starrocks::parquet::FileReader::_init_group_readers()
    @          0x59564b3 starrocks::parquet::FileReader::init()
    @          0x57a4faa starrocks::HdfsParquetScanner::do_open()
    @          0x579a917 starrocks::HdfsScanner::open()
    @          0x5706b27 starrocks::connector::HiveDataSource::_init_scanner()
    @          0x5708cde starrocks::connector::HiveDataSource::open()
    @          0x316db3c starrocks::pipeline::ConnectorChunkSource::_read_chunk()
    @          0x3415c5f starrocks::pipeline::ChunkSource::buffer_next_batch_chunks_blocking()
    @          0x3169e61 _ZZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS_12RuntimeStateEiENKUlvE_clEv
    @          0x3172cee starrocks::workgroup::ScanExecutor::worker_thread()
    @          0x5126bc2 starrocks::ThreadPool::dispatch_thread()
    @          0x51216ba starrocks::thread::supervise_thread()
    @     0x7f242f5d144b start_thread
    @     0x7f242ebc052f __GI___clone
    @                0x0 (unknown)
```

### Steps to reproduce the behavior (Required)
1. Table definition in Glue:
```
CREATE TABLE iceberg_catalog.mydb.events_tb (
  ...
  ts TIMESTAMP,
  ...)
USING iceberg
PARTITIONED BY (days(ts))
LOCATION 's3://bucket/mydb.db/events_tb'
TBLPROPERTIES (
  'format' = 'iceberg/parquet',
  'format-version' = '2',
  'write.delete.mode' = 'merge-on-read',
  'write.merge.mode' = 'merge-on-read',
  'write.metadata.delete-after-commit.enabled' = 'true',
  'write.metadata.previous-versions-max' = '10',
  'write.update.mode' = 'merge-on-read')
```
2. External Iceberg catalog
```
CREATE EXTERNAL CATALOG `iceberg_glue_catalog`
PROPERTIES ("aws.s3.enable_ssl"  =  "false",
"aws.s3.endpoint"  =  "s3.us-east-1.amazonaws.com",
"aws.s3.enable_path_style_access"  =  "true",
"aws.glue.use_instance_profile"  =  "true",
"aws.s3.use_instance_profile"  =  "true",
"aws.s3.region"  =  "us-east-1",
"aws.glue.region"  =  "us-east-1",
"type"  =  "iceberg",
"iceberg.catalog.type"  =  "glue"
)
```
4. Query table from SR
```
SELECT * FROM iceberg_glue_catalog.mydb.events_tb
WHERE ts >= CURRENT_DATE()
```

### Expected behavior (Required)
Query results

### Real behavior (Required)
BE process crashes with error above

### StarRocks version (Required)
3.0.4 f8ff06d

**Comment by Dshadowzh** [comment]
@DorianZheng PTAL

**Comment by Smith-Cruise** [comment]
Can you provide a minimal sql to reproduce it?

Like `spark-sql` create table, insert into values, then using StarRocks query it.

Also, in starrocks 3.0 we don't support iceberg v2 yet, we support it 3.1

**Comment by pedrong** [comment]
This issue was solved on 3.1
Thanks for the help
